nixos/modules/services/web-apps/gancio.nix

```diff
@@ -0,0 +1,280 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+let
+  cfg = config.services.gancio;
+  settingsFormat = pkgs.formats.json { };
+  inherit (lib)
+    mkEnableOption
+    mkPackageOption
+    mkOption
+    types
+    literalExpression
+    mkIf
+    optional
+    mapAttrsToList
+    concatStringsSep
+    concatMapStringsSep
+    getExe
+    mkMerge
+    mkDefault
+    ;
+in
+{
+  options.services.gancio = {
+    enable = mkEnableOption "Gancio, a shared agenda for local communities";
+
+    package = mkPackageOption pkgs "gancio" { };
+
+    plugins = mkOption {
+      type = with types; listOf package;
+      default = [ ];
+      example = literalExpression "[ pkgs.gancioPlugins.telegram-bridge ]";
+      description = ''
+        Paths of gancio plugins to activate (linked under $WorkingDirectory/plugins/).
+      '';
+    };
+
+    user = mkOption {
+      type = types.str;
+      description = "The user (and PostgreSQL database name) used to run the gancio server";
+      default = "gancio";
+    };
```

Comment:
It's standard to expose such an option; if it's broken then that should be fixed.

Reply:
Point me to the standard then. `rg -I 'users.users' nixos/modules/services/web-apps/` tells you that it is very much not *the standard*.

Reply:
I'm agreeing that it should be fixed, why so salty?

Reply:
jbgi: Please consider my comments as a suggestion, not as a change request.

Reply:
the integration test fails when changing user only because of the hardcoded `sudo -u gancio` in https://github.com/NixOS/nixpkgs/pull/279011/files#diff-2f1ee4652f3da848659b72a7b75f4866ad3ef8806b7bafa44caeb9598b7d4de1R78
If modified to the actual cutom user, the test succeed, so I don't think there is any issue with using a customized user for the service.

Reply:
Though I'm now seeing that lots of services only configure an actual system user if the user option is set to the default value. So we should maybe follow this convention?

Reply:
did that in last commit.